packages/@tinacms/mdx/src/parse/parseShortcode.test.ts

```diff
@@ -0,0 +1,87 @@
+/**
+
+Copyright 2021 Forestry.io Holdings, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+import { it, expect, describe } from 'vitest'
+import { parseShortcode } from './parseShortcode'
+
+describe('parseShortcode', () => {
+  describe('with keyed field', () => {
+    it('parses attributes', () => {
+      const result = parseShortcode('{{< signature foo="bar123">}}', {
+        name: 'signature',
+        label: 'Signature',
+        match: {
+          start: '{{<',
+          end: '>}}',
+        },
+        fields: [
+          {
+            name: 'foo',
+            label: 'foo label',
+            type: 'string',
+          },
+        ],
+      })
+      expect(result).toEqual('<signature foo="bar123">\n</signature>')
+    })
+  })
+
+  describe('with unkeyed attributes', () => {
+    it('parses attributes', () => {
+      const result = parseShortcode('{{< signature "bar123" >}}', {
+        name: 'signature',
+        label: 'Signature',
+        match: {
+          start: '{{<',
+          end: '>}}',
+        },
+        fields: [
+          {
+            name: 'text',
+            label: 'text',
+            type: 'string',
+          },
+        ],
+      })
+      expect(result).toEqual('<signature text="bar123">\n</signature>')
+    })
+  })
+
+  describe('with children', () => {
+    it('parses children field', () => {
+      const result = parseShortcode(
+        '{{< signature >}}\n# FOO\n##Bar\n{{< /signature >}}',
+        {
+          name: 'signature',
+          label: 'Signature',
+          match: {
+            start: '{{<',
+            end: '>}}',
+          },
+          fields: [
+            {
+              name: 'children',
+              label: 'children',
+              type: 'string',
```

Comment:
I think we'll want to advise that this should be `rich-text` so `## FOO\n##Bar` are parsed as H2 elements.

Reply:
> This looks great, the only remaining question I'd have which might be nice to cover in the tests is for shortcodes with no fields at all (eg. `{{< ad-panel >}}`. I think it will probably just work, but would clicking on it take you to an empty form list?

Oh right, hadn't considered that. I don't think it's too too weird to have it link to an empty form. Maybe if there's no fields it becomes not clickable at all (if that's not too much more effort)? 

Also, did you have any thoughts on this?:
> Can/should we use a different key for inline raw-strings than "text"? I took a quick crack at modifying that logic but might need @jeffsee55's assistance with that.

I'm just a bit scared that "text" might be too common of a prop name for existing shortcodes so it might conflict.

Reply:
> I'm just a bit scared that "text" might be too common of a prop name for existing shortcodes so it might conflict.

Maybe `_value`? Or something like that?

Also, this might need for of a deep refactor because the name of the field can't contain any dashes, so for example `ad-panel` as the _name_ will break everything. This worked previously because the text field was just open-ended

Reply:
Yeah _value would work! Is that something you'd be able to help with? Not 100%, but I think `text` may be used by mdx logic outside this PR, so I wasn't sure if it would be a trivial change.

> Also, this might need for of a deep refactor because the name of the field can't contain any dashes, so for example ad-panel as the name will break everything. This worked previously because the text field was just open-ended

I have an optional `name` property under the `match` prop that you can use to supply names with special characters like dashes:

```
name: "adPanel",
match: {
  start: "<<%",
  name: "ad-panel",
  end: "%>>
}
// ...
```
outputs: `<<% ad-panel %>>`

Reply:
Ok cool, that works. I might take stab at those changes.